### ISSUE TITLE
Fully switch to Google Analytics 4

### ISSUE THREAD

**Issue by mlbrgl** [owid/owid-grapher#1402]
## Core problem

_On 1 July 2023, Universal Analytics properties will stop processing new hits. If you still rely on Universal Analytics, we recommend that you complete your move to Google Analytics 4._

## Proposed solution

The general GA4 setup should be complete. We probably need to take a closer look at events now.

- [x] reassess events: should they all be ported or only selectively ported?
- [ ] Archive universal analytics data
_You can continue to use and collect new data in your Universal Analytics properties until 1 July 2023. After that, you will be able to access your previously processed data in Universal Analytics for a period of at least six months. We know your data is important to you and we strongly encourage you to export your historical reports during this time._

**Comment by larsyencken** [comment]
@mlbrgl Hey, so I'm a bit confused with our multiple GA setups in parallel. Which one is our reference now?

**Comment by mlbrgl** [comment]
Added [slack reminder](https://owid.slack.com/archives/CQQUA2C2U/p1655215920826419)

**Comment by larsyencken** [comment]
Downgraded prio to `nice to have`

**Comment by larsyencken** [comment]
Enough time passed that it became important to do now, bumped the priority and assigned @marcelgerber for this cycle.

**Comment by marcelgerber** [comment]
So a quick update on this:

One [major difference between UA and GA4 is the hierarchy and metadata for events](https://www.rootandbranchgroup.com/events-in-ga4-vs-ua/):
- In UA, we have event category (required), event action (required), event label and event value as properties for an event.
- In GA4, we have event name as the only required field, and then are pretty free with additional metadata and their field names - they are pretty similar to JSON.

This, in turn, also means that we don't get to use the same hierarchy that we are using right now. For example, right now we have:
- `SiteClick > chart-click-map`, `SiteClick > chart-click-sources`
- `GlobalEntitySelector > close`, `GlobalEntitySelector > change`
- where, for these, the event label is `close`, `change`, etc. respectively.

Going forward, we should use the existing _event category_ as the _event name_, and put the _event action_ and _event label_ into metadata fields.


Also, [GA4 tracks some events automatically](https://support.google.com/analytics/answer/9216061?hl=en) ("enhanced measurement"), without us needing to set this up automatically:
- `page_view`, of course
- `scroll`, which indicates having scrolled to the bottom of the page
- `click`, which is a user clicking away to a different domain

Other, related, auto-trackers we can set up in Google Tag Manager.


Another interesting tidbit is that, since we're using Google Tag Manager (GTM) to set up GA4, it is GTM that is handling our custom events at first, and we need to set up a rule in GTM to forward these events to GA4.

**Comment by marcelgerber** [comment]
Also, in terms of event names (event _categories_ and _actions_), we're a bit all over the place at the moment.
Some are camel case, some are kebab case, some are all-caps snake case.

Luckily, they are all defined in https://github.com/owid/owid-grapher/blob/89151edbbc7b9321629efae15e4043e53745c45c/packages/%40ourworldindata/grapher/src/core/GrapherAnalytics.ts and https://github.com/owid/owid-grapher/blob/89151edbbc7b9321629efae15e4043e53745c45c/site/SiteAnalytics.ts, so it should be relatively easy to change them now. And now is also a good point in time to do just that.

**Comment by marcelgerber** [comment]
TODOs:
- [ ] In GTM, forward all `owid\..*` events to GA4
- [ ] In Netlify, add a script snippet `window.dataLayer = window.dataLayer || []` _before_ the GTM snippet

**Comment by larsyencken** [comment]
Thanks for the update Marcel! I totally agree that this is the moment to standardise our event names. I'm not opinionated on the naming style, as long as it's consistent. We should probably apply the same style to all event fields (i.e. names, actions, and labels)

**Comment by marcelgerber** [comment]
Agreed!
There's more details in the PR #2301, but I now went with the style `owid.event_name_in_snake_case`:
- The `owid` prefix is there because we need to specify in Google Tag Manager (GTM) _which_ events we want to forward to GA4, and having a common prefix makes that easy. Also, it's easy to see that these are custom events we're firing from code, as opposed to GA4's builtins like `page_view`.
- Snake case because that's consistent with GA4's inbuilt events, like `page_view` and `session_start`.

At this moment, event names are consistent, but further event info (`eventAction`) is not; it's mostly in kebab case right now.
